File: packages/maplibre/android/src/main/java/io/capawesome/capacitorjs/plugins/maplibre/classes/options/RemoveLayerByIdOptions.java
package io.capawesome.capacitorjs.plugins.maplibre.classes.options;

import androidx.annotation.NonNull;
import com.getcapacitor.PluginCall;
import io.capawesome.capacitorjs.plugins.maplibre.MapLibreHelper;
import io.capawesome.capacitorjs.plugins.maplibre.classes.CustomExceptions;

public class RemoveLayerByIdOptions {

    @NonNull
    private final String layerId;

    @NonNull
    private final String mapId;

    public RemoveLayerByIdOptions(@NonNull PluginCall call) throws Exception {
        String layerId = call.getString("layerId");
        if (layerId == null) {
            throw CustomExceptions.LAYER_ID_MISSING;
        }
        this.layerId = layerId;
        this.mapId = MapLibreHelper.getMapIdFromCall(call);
    }

    @NonNull
    public String getLayerId() {
        return layerId;
    }

    @NonNull
    public String getMapId() {
        return mapId;
    }
}

File: packages/maplibre/ios/Plugin/Classes/Options/RemoveLayerByIdOptions.swift
import Capacitor
import Foundation

@objc public class RemoveLayerByIdOptions: NSObject {
    let layerId: String
    let mapId: String

    init(_ call: CAPPluginCall) throws {
        self.layerId = try MapLibreHelper.getString(call, "layerId", .layerIdMissing)
        self.mapId = try MapLibreHelper.getString(call, "mapId", .mapIdMissing)
    }
}
